stan/math/prim/fun/value_of.hpp

```diff
@@ -18,8 +18,18 @@ namespace math {
  * @return Forwarded input argument
  **/
 template <typename T, require_st_arithmetic<T>* = nullptr>
-inline decltype(auto) value_of(T&& x) {
-  return std::forward<T>(x);
+inline T value_of(T&& x) {
+  return std::move(x);
+}
+
+template <typename T, require_st_arithmetic<T>* = nullptr>
+inline T& value_of(T& x) {
+  return x;
+}
+
+template <typename T, require_st_arithmetic<T>* = nullptr>
+inline const T& value_of(const T& x) {
+  return x;
```

Comment:
Again no need to write all these out. Replacing `decltype(auto)` with `T` is enough.